app/components/UI/UrlAutocomplete/index.tsx

```diff
@@ -211,15 +213,33 @@ const UrlAutocomplete = forwardRef<
     sourcePage: 'MainView',
   });
 
+  const { trackEvent, createEventBuilder } = useMetrics();
+
   const goToSwaps = useCallback(async (result: TokenSearchResult) => {
     try {
+      let properties: JsonMap;
+      if (latestSearchTerm.current?.startsWith('0x')) {
+        properties = {
+          token_address: latestSearchTerm.current,
+        };
+      } else {
+        properties = {
+          token_symbol: result.symbol,
+        };
+      }
+      trackEvent(
+        createEventBuilder(MetaMetricsEvents.TOKEN_SEARCH_DISCOVERY_TOKEN_SWAP_OPENED)
+        .addProperties(properties)
+        .build()
+      );
+
```

Comment:
C: the whole point of using a builder is to ease building events. You don't have to build manually and then pass it.
Allows to have JsonMap type inferred by addProperties (remove JsonMap import too).
Removes the let: not a risk, but adds cognitive load to reader as they might assume properties can change later, even if it doesn’t.

Just create the builder first then add props as needed.
```suggestion
const eventBuilder = createEventBuilder(MetaMetricsEvents.TOKEN_SEARCH_DISCOVERY_TOKEN_SWAP_OPENED);
eventBuilder.addProperties(
  latestSearchTerm.current?.startsWith('0x')
    ? { token_address: latestSearchTerm.current }
    : { token_symbol: result.symbol },
);
trackEvent(eventBuilder.build());

```